Explication des données

Algorithme LDA

### l'analyse discriminante linéaire permet d'expliquer et de prédire l'appartenance d’un individu à une classe (groupe) prédéfinie à partir de ses caractéristiques mesurées à l’aide de variables prédictives. c'est à dire diviser une matrice de réponse en groupes en fonction d’un facteur en trouvant la combinaison de variables qui donne la meilleure séparation possible entre les groupes.

### Ce modele permet de prédire quels sont individus les plus susceptibles d’avoir ou d’utiliser un compte bancaire sur certains pays d'afrique. Il permet de predire la disponibimté d'avoir un compte ou pas dans certain pays de l'afrique. 


Explication des données

### Ces données sont composées : une liste de pays dont les études ont été fait, de l'années, de leurs locations, la disponibilité de compte banquaire, de leurs situation matrimoniale, de leur sexe, de leur niveau d'étude, du type de travail et de la disponiblté du telephone de portzble  

importation des modules


In [1]:
import pandas as pd 
import numpy as np 
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis

### 1. Load the dataset

In [2]:
# Load files into a pandas dataframe
train = pd.read_csv('Train.csv')
test = pd.read_csv('Test.csv')

In [3]:
train

country  year       uniqueid bank_account location_type  \
0       Kenya  2018     uniqueid_1          Yes         Rural   
1       Kenya  2018     uniqueid_2           No         Rural   
2       Kenya  2018     uniqueid_3          Yes         Urban   
3       Kenya  2018     uniqueid_4           No         Rural   
4       Kenya  2018     uniqueid_5           No         Urban   
...       ...   ...            ...          ...           ...   
23519  Uganda  2018  uniqueid_2113           No         Rural   
23520  Uganda  2018  uniqueid_2114           No         Rural   
23521  Uganda  2018  uniqueid_2115           No         Rural   
23522  Uganda  2018  uniqueid_2116           No         Urban   
23523  Uganda  2018  uniqueid_2117           No         Rural   

      cellphone_access  household_size  age_of_respondent  \
0                  Yes               3                 24   
1                   No               5                 70   
2                  Yes               5                 26   
3                  Yes               5                 34   
4                   No               8                 26   
...                ...             ...                ...   
23519              Yes               4                 48   
23520              Yes               2                 27   
23521              Yes               5                 27   
23522              Yes               7                 30   
23523              Yes              10                 20   

      gender_of_respondent relationship_with_head           marital_status  \
0                   Female                 Spouse  Married/Living together   
1                   Female      Head of Household                  Widowed   
2                     Male         Other relative     Single/Never Married   
3                   Female      Head of Household  Married/Living together   
4                     Male                  Child     Single/Never Married   
...                    ...                    ...                      ...   
23519               Female      Head of Household       Divorced/Seperated   
23520               Female      Head of Household     Single/Never Married   
23521               Female                 Parent                  Widowed   
23522               Female                 Parent       Divorced/Seperated   
23523                 Male                  Child     Single/Never Married   

                       education_level                   job_type  
0                  Secondary education              Self employed  
1                  No formal education       Government Dependent  
2      Vocational/Specialised training              Self employed  
3                    Primary education  Formally employed Private  
4                    Primary education        Informally employed  
...                                ...                        ...  
23519              No formal education               Other Income  
23520              Secondary education               Other Income  
23521                Primary education               Other Income  
23522              Secondary education              Self employed  
23523              Secondary education                  No Income  

[23524 rows x 13 columns]

In [4]:
test

country  year       uniqueid location_type cellphone_access  \
0       Kenya  2018  uniqueid_6056         Urban              Yes   
1       Kenya  2018  uniqueid_6060         Urban              Yes   
2       Kenya  2018  uniqueid_6065         Rural               No   
3       Kenya  2018  uniqueid_6072         Rural               No   
4       Kenya  2018  uniqueid_6073         Urban               No   
...       ...   ...            ...           ...              ...   
10081  Uganda  2018  uniqueid_2998         Rural               No   
10082  Uganda  2018  uniqueid_2999         Urban              Yes   
10083  Uganda  2018  uniqueid_3000         Urban              Yes   
10084  Uganda  2018  uniqueid_3001         Rural              Yes   
10085  Uganda  2018  uniqueid_3002         Urban              Yes   

       household_size  age_of_respondent gender_of_respondent  \
0                   3                 30                 Male   
1                   7                 51                 Male   
2                   3                 77               Female   
3                   6                 39               Female   
4                   3                 16                 Male   
...               ...                ...                  ...   
10081               2                 62               Female   
10082               8                 42                 Male   
10083               1                 39                 Male   
10084               6                 28               Female   
10085               7                 16                 Male   

      relationship_with_head           marital_status  \
0          Head of Household  Married/Living together   
1          Head of Household  Married/Living together   
2                     Parent  Married/Living together   
3          Head of Household  Married/Living together   
4                      Child     Single/Never Married   
...                      ...                      ...   
10081                 Spouse  Married/Living together   
10082      Head of Household  Married/Living together   
10083      Head of Household     Single/Never Married   
10084                 Spouse  Married/Living together   
10085         Other relative     Single/Never Married   

                       education_level                      job_type  
0                  Secondary education  Formally employed Government  
1      Vocational/Specialised training     Formally employed Private  
2                  No formal education          Remittance Dependent  
3                    Primary education          Remittance Dependent  
4                  Secondary education          Remittance Dependent  
...                                ...                           ...  
10081                Primary education                 Self employed  
10082                Primary education                 Self employed  
10083              Secondary education                  Other Income  
10084                Primary education                 Self employed  
10085              Secondary education                  Other Income  

[10086 rows x 12 columns]

In [5]:
# Let’s observe the shape of our datasets.
print('train data shape :', train.shape)
print('test data shape :', test.shape)

train data shape : (23524, 13)
test data shape : (10086, 12)


### normalisation 

In [6]:
# Check for missing values
print('missing values:', train.isnull().sum())

missing values: country                   0
year                      0
uniqueid                  0
bank_account              0
location_type             0
cellphone_access          0
household_size            0
age_of_respondent         0
gender_of_respondent      0
relationship_with_head    0
marital_status            0
education_level           0
job_type                  0
dtype: int64


In [7]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23524 entries, 0 to 23523
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   country                 23524 non-null  object
 1   year                    23524 non-null  int64 
 2   uniqueid                23524 non-null  object
 3   bank_account            23524 non-null  object
 4   location_type           23524 non-null  object
 5   cellphone_access        23524 non-null  object
 6   household_size          23524 non-null  int64 
 7   age_of_respondent       23524 non-null  int64 
 8   gender_of_respondent    23524 non-null  object
 9   relationship_with_head  23524 non-null  object
 10  marital_status          23524 non-null  object
 11  education_level         23524 non-null  object
 12  job_type                23524 non-null  object
dtypes: int64(3), object(10)
memory usage: 2.3+ MB


### on essayer de remplacer les yes et no par des chiffre binaires

In [8]:
#import preprocessing module
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import MinMaxScaler

# Cobvert target label to numerical Data
le = LabelEncoder()
train['bank_account'] = le.fit_transform(train['bank_account'])

#Separate training features from target
X_train = train.drop(['bank_account'], axis=1)
y_train = train['bank_account']

print(y_train)

0        1
1        0
2        1
3        0
4        0
        ..
23519    0
23520    0
23521    0
23522    0
23523    0
Name: bank_account, Length: 23524, dtype: int32


### cette methode permet de labelise les caractere

In [9]:
# function to preprocess our data from train models
def preprocessing_data(data):

    # Convert the following numerical labels from interger to float
    float_array = data[["household_size", "age_of_respondent", "year"]].values.astype(float)
    
    # categorical features to be onverted to One Hot Encoding
    categ = ["relationship_with_head",
             "marital_status",
             "education_level",
             "job_type",
             "country"]
    
    # One Hot Encoding conversion
    data = pd.get_dummies(data, prefix_sep="_", columns=categ)
    
    # Label Encoder conversion
    data["location_type"] = le.fit_transform(data["location_type"])
    data["cellphone_access"] = le.fit_transform(data["cellphone_access"])
    data["gender_of_respondent"] = le.fit_transform(data["gender_of_respondent"])
    
    # drop uniquid column
    data = data.drop(["uniqueid"], axis=1)
    
    # scale our data into range of 0 and 1
    scaler = MinMaxScaler(feature_range=(0, 1))
    data = scaler.fit_transform(data)
    
    return data                  

In [10]:
# preprocess the train data 
processed_train = preprocessing_data(X_train)
processed_test = preprocessing_data(test)

In [11]:
X_Train, X_Val, y_Train, y_val = train_test_split(processed_train, y_train, stratify = y_train, test_size = 0.1, random_state=42)

In [12]:
X_Train

array([[1. , 0. , 0. , ..., 0. , 0. , 1. ],
       [0.5, 1. , 0. , ..., 0. , 1. , 0. ],
       [1. , 0. , 1. , ..., 0. , 0. , 0. ],
       ...,
       [0.5, 1. , 0. , ..., 0. , 1. , 0. ],
       [0. , 1. , 1. , ..., 1. , 0. , 0. ],
       [0. , 0. , 0. , ..., 1. , 0. , 0. ]])

In [13]:
y_Train

23019    0
15717    0
5029     1
21657    0
15193    0
        ..
13297    0
14158    0
18867    0
13529    0
12869    0
Name: bank_account, Length: 21171, dtype: int32

### IMPLEMENTATION DE LA METHODE LOGISTICREGRESSION

In [14]:
# instanciation 
lr=LogisticRegression()
# fitting 
lr.fit(X_Train,y_Train)
# prediction
pred = lr.predict(X_Val)
pred

c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


array([0, 0, 0, ..., 0, 0, 1])

In [15]:
acc = accuracy_score(y_val, pred)
acc

0.8878028049298767

In [16]:
from sklearn.model_selection import GridSearchCV

In [17]:
## parameter du grid
params=[
     {'intercept_scaling':[1,2,3,4,5,5,6,7,8,9,10,12]},
    {'max_iter':[2, 4, 6, 8, 9, 10, 5]}
]
opt_lr=GridSearchCV(lr,params,cv=5,verbose=4)
opt_lr.fit(X_Train,y_Train)

Fitting 5 folds for each of 19 candidates, totalling 95 fits


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=1;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=1;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=1;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=1;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=1;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=2;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=2;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=2;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=2;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=2;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=3;, score=0.880 total time=   0.3s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=3;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=3;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=3;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=3;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=4;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=4;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=4;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=4;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=4;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=5;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=5;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=5;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=5;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=5;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=5;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=5;, score=0.882 total time=   0.3s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=5;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=5;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=5;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=6;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=6;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=6;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=6;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=6;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=7;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=7;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=7;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=7;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=7;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=8;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=8;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=8;, score=0.887 total time=   0.3s
[CV 4/5] END ...............intercept_scaling=8;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=8;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ...............intercept_scaling=9;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ...............intercept_scaling=9;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ...............intercept_scaling=9;, score=0.887 total time=   0.2s
[CV 4/5] END ...............intercept_scaling=9;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ...............intercept_scaling=9;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ..............intercept_scaling=10;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ..............intercept_scaling=10;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ..............intercept_scaling=10;, score=0.887 total time=   0.2s
[CV 4/5] END ..............intercept_scaling=10;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 5/5] END ..............intercept_scaling=10;, score=0.883 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 1/5] END ..............intercept_scaling=12;, score=0.880 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 2/5] END ..............intercept_scaling=12;, score=0.882 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


[CV 3/5] END ..............intercept_scaling=12;, score=0.887 total time=   0.2s
[CV 4/5] END ..............intercept_scaling=12;, score=0.889 total time=   0.2s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check

[CV 5/5] END ..............intercept_scaling=12;, score=0.883 total time=   0.2s
[CV 1/5] END ........................max_iter=2;, score=0.859 total time=   0.0s
[CV 2/5] END ........................max_iter=2;, score=0.859 total time=   0.0s
[CV 3/5] END ........................max_iter=2;, score=0.859 total time=   0.0s
[CV 4/5] END ........................max_iter=2;, score=0.859 total time=   0.0s
[CV 5/5] END ........................max_iter=2;, score=0.859 total time=   0.0s
[CV 1/5] END ........................max_iter=4;, score=0.867 total time=   0.0s
[CV 2/5] END ........................max_iter=4;, score=0.871 total time=   0.0s
[CV 3/5] END ........................max_iter=4;, score=0.868 total time=   0.0s
[CV 4/5] END ........................max_iter=4;, score=0.872 total time=   0.0s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check

[CV 5/5] END ........................max_iter=4;, score=0.869 total time=   0.0s
[CV 1/5] END ........................max_iter=6;, score=0.879 total time=   0.0s
[CV 2/5] END ........................max_iter=6;, score=0.879 total time=   0.0s
[CV 3/5] END ........................max_iter=6;, score=0.880 total time=   0.0s
[CV 4/5] END ........................max_iter=6;, score=0.880 total time=   0.0s
[CV 5/5] END ........................max_iter=6;, score=0.876 total time=   0.0s
[CV 1/5] END ........................max_iter=8;, score=0.881 total time=   0.0s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check

[CV 2/5] END ........................max_iter=8;, score=0.881 total time=   0.0s
[CV 3/5] END ........................max_iter=8;, score=0.885 total time=   0.0s
[CV 4/5] END ........................max_iter=8;, score=0.887 total time=   0.0s
[CV 5/5] END ........................max_iter=8;, score=0.881 total time=   0.0s
[CV 1/5] END ........................max_iter=9;, score=0.881 total time=   0.0s
[CV 2/5] END ........................max_iter=9;, score=0.880 total time=   0.0s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check

[CV 3/5] END ........................max_iter=9;, score=0.887 total time=   0.0s
[CV 4/5] END ........................max_iter=9;, score=0.888 total time=   0.0s
[CV 5/5] END ........................max_iter=9;, score=0.880 total time=   0.0s
[CV 1/5] END .......................max_iter=10;, score=0.881 total time=   0.0s
[CV 2/5] END .......................max_iter=10;, score=0.882 total time=   0.0s
[CV 3/5] END .......................max_iter=10;, score=0.888 total time=   0.0s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check

[CV 4/5] END .......................max_iter=10;, score=0.888 total time=   0.0s
[CV 5/5] END .......................max_iter=10;, score=0.882 total time=   0.0s
[CV 1/5] END ........................max_iter=5;, score=0.874 total time=   0.0s
[CV 2/5] END ........................max_iter=5;, score=0.877 total time=   0.0s
[CV 3/5] END ........................max_iter=5;, score=0.875 total time=   0.0s
[CV 4/5] END ........................max_iter=5;, score=0.879 total time=   0.0s
[CV 5/5] END ........................max_iter=5;, score=0.872 total time=   0.0s


c:\Python\Python310\lib\site-packages\sklearn\linear_model\_logistic.py:444: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


GridSearchCV(cv=5, estimator=LogisticRegression(),
             param_grid=[{'intercept_scaling': [1, 2, 3, 4, 5, 5, 6, 7, 8, 9,
                                                10, 12]},
                         {'max_iter': [2, 4, 6, 8, 9, 10, 5]}],
             verbose=4)

In [18]:
print(opt_lr.best_estimator_)
print(opt_lr.best_params_)
print(opt_lr.best_score_)

LogisticRegression()
{'intercept_scaling': 1}
0.8843230853399617


## implementation du modele KNN

In [23]:
### 2 application d'un modele 
## appel de la fonction 
from sklearn.neighbors import KNeighborsClassifier ## le modele
### 1 instanciation du modele
model_knn= KNeighborsClassifier(n_neighbors=5)
### 2 fitting du modele
model_knn.fit(X_Train,y_Train)
### 3 prediction 
pred = model_knn.predict(X_Val)
pred

array([0, 0, 0, ..., 0, 0, 1])

In [24]:
## parameter du grid
params=[
    {'n_neighbors':[1,2,3,4,5,5,6,7,8,9,10,12]},
    {'leaf_size':[2, 4, 6, 8, 9, 10, 5]}
]
# initiation
model_knn=KNeighborsClassifier()
test_model = GridSearchCV(model_knn,params,cv=5,verbose=10)
test_model.fit(X_Train,y_Train)

Fitting 5 folds for each of 19 candidates, totalling 95 fits
[CV 1/5; 1/19] START n_neighbors=1..............................................
[CV 1/5; 1/19] END ...............n_neighbors=1;, score=0.838 total time=   0.5s
[CV 2/5; 1/19] START n_neighbors=1..............................................
[CV 2/5; 1/19] END ...............n_neighbors=1;, score=0.833 total time=   0.6s
[CV 3/5; 1/19] START n_neighbors=1..............................................
[CV 3/5; 1/19] END ...............n_neighbors=1;, score=0.838 total time=   0.5s
[CV 4/5; 1/19] START n_neighbors=1..............................................
[CV 4/5; 1/19] END ...............n_neighbors=1;, score=0.833 total time=   0.5s
[CV 5/5; 1/19] START n_neighbors=1..............................................
[CV 5/5; 1/19] END ...............n_neighbors=1;, score=0.840 total time=   0.4s
[CV 1/5; 2/19] START n_neighbors=2..............................................
[CV 1/5; 2/19] END ...............n_neighbors=2;

GridSearchCV(cv=5, estimator=KNeighborsClassifier(),
             param_grid=[{'n_neighbors': [1, 2, 3, 4, 5, 5, 6, 7, 8, 9, 10,
                                          12]},
                         {'leaf_size': [2, 4, 6, 8, 9, 10, 5]}],
             verbose=10)

In [26]:
print(test_model.best_estimator_)
print(test_model.best_params_)
print(test_model.best_score_)

KNeighborsClassifier(n_neighbors=12)
{'n_neighbors': 12}
0.8841811299878033


## Implementation de la methode Arbre de decision

In [32]:
# arbre de decision 
# instanciation du modele
from sklearn.tree import DecisionTreeClassifier
model_dtc= DecisionTreeClassifier()
# fitting
model_dtc.fit(X_Train, y_Train)
# prediction
pred_dtc = model_dtc.predict(X_Val)
pred_dtc

array([0, 1, 1, ..., 0, 0, 1])

In [33]:
## parameter du grid
params=[
     
    {'max_depth':[2, 4, 6, 8, 9, 10, 5]}
]
# instanciation
opt_dtc = GridSearchCV(model_dtc, params, cv=15, verbose=10)
opt_dtc.fit(X_Train, y_Train)

Fitting 15 folds for each of 7 candidates, totalling 105 fits
[CV 1/15; 1/7] START max_depth=2................................................
[CV 1/15; 1/7] END .................max_depth=2;, score=0.876 total time=   0.0s
[CV 2/15; 1/7] START max_depth=2................................................
[CV 2/15; 1/7] END .................max_depth=2;, score=0.878 total time=   0.0s
[CV 3/15; 1/7] START max_depth=2................................................
[CV 3/15; 1/7] END .................max_depth=2;, score=0.877 total time=   0.0s
[CV 4/15; 1/7] START max_depth=2................................................
[CV 4/15; 1/7] END .................max_depth=2;, score=0.878 total time=   0.0s
[CV 5/15; 1/7] START max_depth=2................................................
[CV 5/15; 1/7] END .................max_depth=2;, score=0.880 total time=   0.0s
[CV 6/15; 1/7] START max_depth=2................................................
[CV 6/15; 1/7] END .................max_depth=2

GridSearchCV(cv=15, estimator=DecisionTreeClassifier(),
             param_grid=[{'max_depth': [2, 4, 6, 8, 9, 10, 5]}], verbose=10)

In [34]:
print(opt_dtc.best_estimator_)
print(opt_dtc.best_params_)
print(opt_dtc.best_score_)

DecisionTreeClassifier(max_depth=8)
{'max_depth': 8}
0.8831892475752033


## Implementation de la methode LDA

In [29]:
# INSTANCIATION
lda = LinearDiscriminantAnalysis()
# fitting
lda.fit(X_Train,y_Train)
# prediction
pred = lda.predict(X_Val)
pred

array([0, 0, 0, ..., 0, 0, 1])

In [20]:
acc = accuracy_score(y_val, pred)
acc

0.8835529111772206

In [21]:
## parameter du grid
params=[
     {'solver':['svd', 'lsqr', 'eigen' ] },
    {'shrinkage':[2, 4, 6, 8, 9, 10, 5]}
]
opt_lr=GridSearchCV(lda,params,cv=5,verbose=4)
opt_lr.fit(X_Train,y_Train)

Fitting 5 folds for each of 10 candidates, totalling 50 fits
[CV 1/5] END ........................solver=svd;, score=0.878 total time=   0.0s
[CV 2/5] END ........................solver=svd;, score=0.877 total time=   0.0s
[CV 3/5] END ........................solver=svd;, score=0.885 total time=   0.0s
[CV 4/5] END ........................solver=svd;, score=0.883 total time=   0.0s
[CV 5/5] END ........................solver=svd;, score=0.877 total time=   0.0s
[CV 1/5] END .......................solver=lsqr;, score=0.880 total time=   0.2s
[CV 2/5] END .......................solver=lsqr;, score=0.877 total time=   0.0s
[CV 3/5] END .......................solver=lsqr;, score=0.884 total time=   0.0s
[CV 4/5] END .......................solver=lsqr;, score=0.877 total time=   0.0s
[CV 5/5] END .......................solver=lsqr;, score=0.872 total time=   0.0s
[CV 1/5] END ........................solver=eigen;, score=nan total time=   0.5s
[CV 2/5] END ........................solver=eige

c:\Python\Python310\lib\site-packages\sklearn\model_selection\_validation.py:378: FitFailedWarning: 
40 fits failed out of a total of 50.
The score on these train-test partitions for these parameters will be set to nan.
If these failures are not expected, you can try to debug them by setting error_score='raise'.

Below are more details about the failures:
--------------------------------------------------------------------------------
3 fits failed with the following error:
Traceback (most recent call last):
  File "c:\Python\Python310\lib\site-packages\sklearn\model_selection\_validation.py", line 686, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "c:\Python\Python310\lib\site-packages\sklearn\discriminant_analysis.py", line 605, in fit
    self._solve_eigen(
  File "c:\Python\Python310\lib\site-packages\sklearn\discriminant_analysis.py", line 445, in _solve_eigen
    evals, evecs = linalg.eigh(Sb, Sw)
  File "c:\Python\Python310\lib\site-packages\scipy\li

GridSearchCV(cv=5, estimator=LinearDiscriminantAnalysis(),
             param_grid=[{'solver': ['svd', 'lsqr', 'eigen']},
                         {'shrinkage': [2, 4, 6, 8, 9, 10, 5]}],
             verbose=4)

In [22]:
print(opt_lr.best_estimator_)
print(opt_lr.best_params_)
print(opt_lr.best_score_)

LinearDiscriminantAnalysis()
{'solver': 'svd'}
0.8799301990576092


##### comparons ce modele avec les autres modele

### LDA
LinearDiscriminantAnalysis()
{'solver': 'svd'}
SCORE : 0.8799301990576092

### REGRESSION LOGISTIC
LogisticRegression()
{'intercept_scaling': 1}
SCORE : 0.8843230853399617

### KNN
KNeighborsClassifier(n_neighbors=12)
{'n_neighbors': 12}
SCORE : 0.8841811299878033
### ARBRE DE DECISION
DecisionTreeClassifier(max_depth=8)
{'max_depth': 8}
SCORE : 0.8831892475752033

D'apres les resultats optenu, nous pouvons dire les modules classiques ont un score plus élevé que le modéle LDA, c'est à dire ces modéles sont plus éfficaces que le modéle LDA qui ont un score de 0.88 par compte celui de LDA qui a un score de 0.87 
D'aprés les résultats que nous avons étudier pour faire la prediction des personnes susceptible d'avoir ou pas un compte banquaire

